samples/CMakeLists.txt

```diff
@@ -53,6 +53,33 @@ function(add_hipcl_binary EXEC_NAME)
 
 endfunction()
 
+# ARGN = sources
+function(add_hipcl_device_binary BIN_NAME)
+    set(SOURCES ${ARGN})
+
+    set(BIN_NAME_OBJ "${BIN_NAME}_o")
+
+    add_library("${BIN_NAME_OBJ}" OBJECT ${SOURCES})
+
+    set_source_files_properties(${SOURCES} PROPERTIES LANGUAGE CXX)
+
+    target_link_libraries("${BIN_NAME_OBJ}" "${SANITIZER_LIBS}" "hipcl")
+
+    target_compile_options("${BIN_NAME_OBJ}" PRIVATE "--cuda-device-only")
```

Comment:
Having --cuda-device-only is confusing here. I know that HIPAMD uses this flag to produce module object code but it's hidden behind hipcc and users provide --genco flag instead. The hipcc recognizes this flag and translates it to  --cuda-device-only for HIP-Clang. I suggest adding hipcc counterpart for HIPCL.

Reply:
I initially had the same reaction than you. But I thought about it more and reflected that the goal here is to have `hipcc` be the driver behind the unified HIPCL and HIPLZ, so I'd rather not diverge from AMD's behavior too much.

Reply:
I had in my mind that we should add `hipcc` (not `hipclcc`) that fills its role as a compiler frontend for HIPCL for starters and later have it replaced with HIP-common’s `hipcc`.

Reply:
Fair enough. It's not a big deal to have the --cuda-device-only for a bit until the adoption of `hipcc`.